src/tabpfn_common_utils/telemetry/core/decorators.py

```diff
@@ -16,6 +20,190 @@
 # Logger
 logger = logging.getLogger(__name__)
 
+# Current extension
+_CONTEXT_VARS = {}
+
+
+def _get_context_var():
+    """Get the shared context variable, ensuring it's the same 
+    instance across all imports.
+    
+    Returns:
+        The context variable.
+    """
+    var_name = "tabpfn_current_extension"
+    if var_name not in _CONTEXT_VARS:
+        _CONTEXT_VARS[var_name] = contextvars.ContextVar[Optional[str]](
```

Comment:
Here I'm not sure I got why using the dictionary instead of defining a global var directly, say `current_extension = contextvars.ContextVar...`?